pkg/operator/ceph/object/objectstore.go

```diff
@@ -411,14 +407,12 @@ func createMultisite(objContext *Context, endpointArg string) error {
 		}
 	}
 
-	if updatePeriod {
-		// the period will help notify other zones of changes if there are multi-zones
-		_, err := runAdminCommand(objContext, false, "period", "update", "--commit")
-		if err != nil {
-			return errorOrIsNotFound(err, "failed to update period")
-		}
-		logger.Debugf("updated period for realm %v", objContext.Realm)
+	// the period will help notify other zones of changes if there are multi-zones
+	_, err = runAdminCommand(objContext, false, "period", "update", "--commit")
```

Comment:
I thought there were side effects of updating the period with every reconcile when nothing changed. But that was a long time ago, so maybe something changed. Did the rgw team recommend this change? Or at least would be good to review the expectations around updating the period.

Reply:
I'm not sure. It would be good to get clarity on this. @thotz ?

From what I observed, the command didn't fail, but that doesn't mean there couldn't be a side effect.

Reply:
I am not aware of side effects wrt `period update` command. May @alimaredia or @cbodley have a more clear idea.
Even if there is no side effect IMHO it is better to check the via `period get` to see whether zone/zonegroups are synced before performing the `period commit` than performing everytime

Reply:
each `period commit` would generate a new period 'epoch', and all radosgws in that realm would have to pause incoming requests to re-initialize themselves with that new period configuration. that should be avoided if nothing actually changed

Reply:
@cbodley is there a straightforward way to know whether we should commit a new period or not? It's going to be tricky to inspect each zone, group, realm and decide to update or not. 

Also, looking ahead, is it a reasonable feature request for the RGW to make the commands idempotent and not update the RGWs if there are no changes needed?

Reply:
> @cbodley is there a straightforward way to know whether we should commit a new period or not? It's going to be tricky to inspect each zone, group, realm and decide to update or not.

so rook doesn't know whether or not it changed anything? you could presumably run `period update` without --commit to generate the staging period, then compare the json output of `period get` (the current period) with `period get --staging` (the updated period) to decide whether or not to `period commit`

> Also, looking ahead, is it a reasonable feature request for the RGW to make the commands idempotent and not update the RGWs if there are no changes needed?

the command is doing what you told it to do, so i'm not crazy about the idea. but if someone wants to work on that, i'm willing to help with review

Reply:
I appreciate the tip about the staging period. For now, I reverted back to our old method of seeing if we needed updates with an addition. It seems that some of our code around multisite configuration always updates realms/zonegroups/zones/periods, and we will have to modify those in the future. The `staging` feature should help us do that more easily.